mlflow/models/evaluation/base.py

```diff
@@ -1247,6 +1247,44 @@ def predict(self, context, model_input: pd.DataFrame):
     return _PythonModelPyfuncWrapper(python_model, None, None)
 
 
+def _get_model_from_deployment_endpoint_uri(
+    endpoint_uri: str, params: Optional[Dict[str, Any]] = None
+):
+    from mlflow.metrics.genai.model_utils import _call_deployments_api, _parse_model_uri
+    from mlflow.pyfunc.model import _PythonModelPyfuncWrapper
+
+    class ModelFromDeploymentEndpoint(mlflow.pyfunc.PythonModel):
+        def __init__(self, endpoint, params):
+            self.endpoint = endpoint
+            self.params = params
+
+        def predict(self, context, model_input: pd.DataFrame):
+            if "inputs" not in model_input.columns:
+                raise MlflowException(
+                    f"Invalid input column: {model_input.columns}. The input column for "
+                    "evaluating an MLflow deployment endpoint must be named 'inputs'"
+                )
+
+            predictions = []
+            for input_data in model_input["inputs"]:
+                if not isinstance(input_data, str):
```

Comment:
Why couldn't the input data be a list of dictionaries corresponding to the OpenAI "messages" input? Seems necessary to support this if we want users to be able to evaluate an endpoint on a dataset of message histories, which is common for chatbots.

Reply:
The main intention here is to start with simple interface and also consistent with OpenAI model flavors. But I agree we can support here (with clear documentation).

Reply:
Got it. Yeah, we should support message histories and include an example :)

Reply:
Dictionary inputs needs some patching for dataset digest computation logic. I think it's not too dirty so ok, but otherwise we might want to leave this as follow-up.